Add back solder_paste as a layer function

gingermusketeer opened the pull request:
This was accidentally removed in the previous PR.

Can someone check the attributes make sense for this function. Please check the attributes that make sense and I can remove any unchecked attributes (I have prechecked the required attributes for each layer):
- [x] order
- [x] uuid
- [x] name
- [x] function
- [ ] sections
- [x] materials

Removed after discussion with Andreas:
- [ ] flexible
- [ ] thickness
- [ ] tolerance_minus
- [ ] tolerance_plus
- [ ] coverage

gingermusketeer committed:
Add back solder_paste as a layer function

Why: It was accidentally removed in the previous PR.

tojanpedersen commented:
This was accidentally removed in the previous PR.

Can someone check the attributes make sense for this function. Please check
the attributes that make sense and I can remove any unchecked attributes (I
have prechecked the required attributes for each layer):

   -  order
   - uuid
   -  name
   -  function
   -  sections
   -  flexible
   -  materials
   -  thickness
   -  tolerance_minus
   -  tolerance_plus
   -  coverage

I suggest this - solderpaste is for us a digital file that we shall send to
customer - it is a part of the PCB spec and will appear as layer 1 and n

Jan


Jan Pedersen
Senior Technical Advisor
[image: Elmatica AS] <http://www.elmatica.com/>
Grensen 12 | N-0159 OSLO | Norway
PHONE +47 91359435 | MOBILE +47 91359435

website <http://www.elmatica.com/> | vCard
<http://www.dynasend.com/signatures/vcard/jan.pedersen-at-elmatica.com.vcf>
| knowledge base <https://elmatica.zendesk.com/forums> | email
<jan.pedersen@elmatica.com> [image: LinkedIn]
<https://www.linkedin.com/company/1377761?trk=tyah&trkInfo=clickedVertical%3Acompany%2CclickedEntityId%3A1377761%2Cidx%3A1-1-1%2CtarId%3A1479030194220%2Ctas%3Aelmati>

*DISCLAIMER: * This email may contain propriety information which is for
the intended recipient only. If an addressing or transmission error has
caused a misdirection, please notify the author by replying to this email.
If you are not the intended recipient you must delete and subsequently not
use, disclose, distribute, copy, print or rely on this email.

Although all emails and attachments are scanned for viruses, Elmatica does
not accept responsibility for any computer virus transmitted by this email.


On Fri, Sep 21, 2018 at 8:52 AM Max Brosnahan <notifications@github.com>
wrote:

> This was accidentally removed in the previous PR.
>
> Can someone check the attributes make sense for this function. Please
> check the attributes that make sense and I can remove any unchecked
> attributes (I have prechecked the required attributes for each layer):
>
>    - order
>    - uuid
>    - name
>    - function
>    - sections
>    - flexible
>    - materials
>    - thickness
>    - tolerance_minus
>    - tolerance_plus
>    - coverage
>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/CircuitData/CircuitData-Language/pull/19
> Commit Summary
>
>    - Add back solder_paste as a layer function
>
> File Changes
>
>    - *M* schema/v1/ottp_circuitdata_schema_products.json
>    <https://github.com/CircuitData/CircuitData-Language/pull/19/files#diff-0>
>    (24)
>
> Patch Links:
>
>    - https://github.com/CircuitData/CircuitData-Language/pull/19.patch
>    - https://github.com/CircuitData/CircuitData-Language/pull/19.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CircuitData/CircuitData-Language/pull/19>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AgJ-nKAPZSfdH6AxYjzXkfWacdt8yB3zks5udIywgaJpZM4WzjSm>
> .
>

gingermusketeer committed:
Remove attributes that are not relevant

gingermusketeer commented:
@tojanpedersen I have updated this PR with some changes that were made after discussion with Andreas. Let me know what you think!

alydersen commented:
I don't see why we wouldn't set the sections. I guess it should cover all sections and should as such be validated to make sure that it covers them all?

tojanpedersen commented:
Allways all...

Jan Pedersen
Senior Technical Advisor

Grensen 12 | N-0159 OSLO | Norway 
PHONE +47 91359435 | MOBILE +47 91359435
  	
website | vCard | knowledge base | email	

DISCLAIMER:  This email may contain propriety information which is for the intended recipient only. If an addressing or transmission error has caused a misdirection, please notify the author by replying to this email. If you are not the intended recipient you must delete and subsequently not use, disclose, distribute, copy, print or rely on this email.

Although all emails and attachments are scanned for viruses, Elmatica does not accept responsibility for any computer virus transmitted by this email. 


> 21. sep. 2018 kl. 13:15 skrev alydersen <notifications@github.com>:
> 
> I don't see why we wouldn't set the sections. I guess it should cover all sections and should as such be validated to make sure that it covers them all?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

gingermusketeer commented:
Here is my take on keeping vs removing sections:

Keeping sections means that
 - We need to validate that they are correct.
 - We can treat them like any layer and use the section data to visualise etc.

Removing them means that
- We need to imply that it applies to all sections.

I was initially in agreement with the first option but I think that as this is intended to be used by lots of people we should ideally have it designed in such a way that people cannot get it wrong. So I am now in the remove camp. @alydersen what are your thoughts?

gingermusketeer committed:
Remove sections attribute for solder_paste

Why: It always applies to all layers and including it means that it needs to be populated correctly. Removing it means that it is implied.

gingermusketeer commented:
@tojanpedersen Updated. Let me know what you think.
